base_layer/mmr/src/backend.rs

```diff
@@ -39,6 +39,16 @@ pub trait ArrayLike {
     /// Return the item at the given index. Use this if you *know* that the index is valid. Requesting a hash for an
     /// invalid index may cause the a panic
     fn get_or_panic(&self, index: usize) -> &Self::Value;
+
+    /// Shortens the array, keeping the first len elements and dropping the rest. If this feature is not supported,
+    /// the function should return `Err(MerkleMountainRangeError:NotSupported)`
+    fn truncate(&mut self, _len: usize) -> Result<(), MerkleMountainRangeError> {
+        Err(MerkleMountainRangeError::NotSupported)
+    }
+
+    /// Execute the given closure for each value in the array
+    fn for_each<F>(&self, f: F) -> Result<(), MerkleMountainRangeError>
```

Comment:
This seems very specific to a call in the ChangeTracker. Perhaps it should live in its own interface, or perhaps only be implemented for `ArrayLike` of `Result<_>`?

Reply:
I like the different trait idea. Negates need for stubbed "NotSupported" methods. Implemented.